File: Day-11: Greedy Algorithm/1) N meetings in one room.cpp
class Solution {
  public:
    int maxMeetings(vector<int>& start, vector<int>& end) {
        int n = start.size();
        vector<vector<int>> nums;
        for(int i = 0; i < n; i++)
        {
            nums.push_back({end[i], start[i], i+1});
        }
        sort(nums.begin(), nums.end());
        int last = -1;
        int ans = 0;
        for(int i = 0; i < n; i++)
        {
            if(nums[i][1] > last)
            {
                ans++;
                last = nums[i][0];
            }
        }
        return ans;
    }
};

File: Day-11: Greedy Algorithm/1) N meetings in one room.java
class Solution {
    public int maxMeetings(int start[], int end[]) {
        int n = start.length;
        int[][] meetings = new int[n][2];
        for(int i = 0; i < n; i++)
        {
            meetings[i][0] = end[i];
            meetings[i][1] = start[i];
        }
        Arrays.sort(meetings, (a, b) -> Integer.compare(a[0], b[0]));
        int count = 0;
        int lastEndTime = -1;
        for (int i = 0; i < n; i++)
        {
            if(meetings[i][1] > lastEndTime)
            {
                count++;
                lastEndTime = meetings[i][0];
            }
        }
        return count;
    }
}
